community/contribute/styleguides/ansible_styleguide.md

```diff
@@ -0,0 +1,156 @@
+
+Blocks, roles or playbooks are never executed in a privileged mode.
+
+We always insert the become directive between the name of a task and the task itself. This also applies to related directives
+like *become_user* or *become_flags*. This is for better visibility if a task is privileged or not.
+
+```yaml
+- name: Copy hddtemp configuration file
+    become: true
+      ansible.builtin.copy:
+      src: "{{ ansible_os_family }}/hddtemp"
+      dest: "{{ hddtemp_conf_file }}"
+      owner: root
+      group: root
+      mode: 0644
+    notify: Restart hddtemp service
+```
+
+### Position of the when condition
+
+If you need to use the when condition please add this at the end-section from the task where it is needed. This makes the code
+easier to understand for others. Ansible lint provides the when condition under the task name for blocks. To keep the code clear
+we decided against it. Please disable this with a noqa if necessary. For example:
+
+```yaml
+- name: "Archive existing {{ resolvconf_file }} file"
+    become: true
+    ansible.posix.synchronize:
+      src: "/etc/resolv.conf"
+      dest: "/etc/resolv.conf.{{ ansible_date_time.date }}"
+      archive: true
+    delegate_to: "{{ inventory_hostname }}"
+    when: stat_resolvconf_file.stat.islnk is defined and not stat_resolvconf_file.stat.islnk
+```
+
+## Usage of collections
+
+Collections are always defined as in the following example.
+
+**netbox.netbox** is here the collection that is used.
+
+```yaml
+- name: Configure netbox manufacturers
+  netbox.netbox.netbox_manufacturer:
+    netbox_url: "{{ netbox_url }}"
+    netbox_token: "{{ netbox_token }}"
+    data:
+      name: "{{ item.value.name }}"
+      slug: "{{ item.value.slug }}"
+      description: "{{ item.value.description | default('') }}"
+    state: present
+  with_dict: "{{ netbox_data_manufacturers }}"
+```
+
+Please don´t declare it in this way!:
+
+```yaml
+collections:
+    - netbox.netbox
+
+  tasks:
+    - name: Manage Discworld site
+      netbox_site:
+        netbox_url: "{{ netbox_url }}"
+        netbox_token: "{{ netbox_token }}"
+        validate_certs: false
+        data:
+          name: Discworld
+          slug: discworld
+        state: present
+```
+
+If you have to use collections please define them in a requirements.yml.
+
+Example yaml:
+
+```yaml
+roles:
+  - name: geerlingguy.certbot
+    version: master
+    type: git
+    src: git+https://github.com/geerlingguy/ansible-role-certbot.git
+...
+
+collections:
+  - name: ansible.netcommon
+    source: https://galaxy.ansible.com
+
+  - name: https://github.com/ansible-collections/ansible.posix.git
+    type: git
+    version: main
+```
+
+## Usage of roles from other collections
+
+If you want to reuse roles please do it in the following way:
+
+First you have to declare the role or collection in the requirements.yml like shown in the example before.
+
+Than you can use it in plybooks like this
```

Comment:
```suggestion
Than you can use it in playbooks like this
```

Reply:
Typo is done. There are more parameters not only when and become. Like changed_when or collections. The most things in this styleguide describes how parameters should be used or placed. I think the example codeblocks are enough for visibility.